src/main/java/com/example/provider/TripServlet.java

```diff
@@ -104,51 +100,11 @@ public TripServlet(
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     ServletUtils.setStandardResponseHeaders(response);
-
-    String tripId = "";
-
     if (isNullOrEmpty(request.getPathInfo()) || request.getPathInfo().equals("/")) {
-      if (servletState.hasNoTrips()) {
```

Comment:
The provider before made the assumption that there would always be a singleton trip/vehicle, so we allowed calling 'GET Trip' without parameters.

This bit of code was added when 'B2B' was introduced to keep backwards compatibility while all clients were migrated to pass a 'Trip id' to 'TRIP GET'. 

I am removing this as part of this CL to prevent having to update the `triggerMatch` call to receive a `Vehicle`. Also happy to split that bit to a separate PR if we think that is better.